openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/ballerinatypegenerators/ArrayTypeGenerator.java

```diff
@@ -90,9 +93,11 @@ public TypeDescriptorNode generateTypeDescriptorNode() throws BallerinaOpenApiEx
                         "maximum ballerina array length.");
             }
         }
-        ArrayTypeDescriptorNode arrayTypeDescriptorNode = createArrayTypeDescriptorNode(
-                typeDescriptorNode, createToken(OPEN_BRACKET_TOKEN), arrayLengthToken,
-                createToken(CLOSE_BRACKET_TOKEN));
+        ArrayDimensionNode dimensionNode = NodeFactory.createArrayDimensionNode(
+                createToken(SyntaxKind.OPEN_BRACKET_TOKEN), arrayLengthToken,
+                createToken(SyntaxKind.CLOSE_BRACKET_TOKEN));
+        NodeList<ArrayDimensionNode> nodeList = createNodeList(dimensionNode);
+        ArrayTypeDescriptorNode arrayTypeDescriptorNode = createArrayTypeDescriptorNode(typeDescriptorNode, nodeList);
```

Comment:
Is there any possibility of `typeDescriptorNode` itself being an array type desc?. If so we need to flatten the dimensions like in https://github.com/ballerina-platform/ballerina-lang/blob/44f9bf90fbaac670eb7637c267fc4190b4b217ec/misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java#L375

Reply:
Yeah, we can have array type desc as its item type, and we handle it through our main type extraction function from schema. 
As an example we have to handle a three-dimensional [array](https://github.com/ballerina-platform/openapi-connectors/blob/bd200dfe506abc68031c40effbd66d60b8c490ae/openapi/graphhopper.directions/openapi.yml#L2827) for some APIs, currently, we generate this [three-dimensional array](https://github.com/ballerina-platform/openapi-connectors/blob/bd200dfe506abc68031c40effbd66d60b8c490ae/openapi/graphhopper.directions/types.bal#L159) for the example. 🙂

Reply:
Then we need to flatten the dimensions. Simply extract the dimensions of inner type desc and add those dimensions to the list of the final array. ie (a[m])[n] ----> a([m][n]).

Reply:
Yeah,  Please check 87c8d53  apply dimension flatten for the nested array.